editor/plugins/path_2d_editor_plugin.cpp

```diff
@@ -222,13 +224,18 @@ bool Path2DEditor::forward_gui_input(const Ref<InputEvent> &p_event) {
 					// N/A, handled in above condition.
 					break;
 
-				case ACTION_MOVING_POINT: {
+				case ACTION_MOVING_POINT:
+				case ACTION_MOVING_NEW_POINT: {
 					undo_redo->create_action(TTR("Move Point in Curve"));
 					undo_redo->add_do_method(curve.ptr(), "set_point_position", action_point, cpoint);
-					undo_redo->add_undo_method(curve.ptr(), "set_point_position", action_point, moving_from);
-					undo_redo->add_do_method(canvas_item_editor, "update_viewport");
-					undo_redo->add_undo_method(canvas_item_editor, "update_viewport");
-					undo_redo->commit_action();
+					if (action == ACTION_MOVING_POINT) {
+						undo_redo->add_undo_method(curve.ptr(), "set_point_position", action_point, moving_from);
+						undo_redo->add_do_method(canvas_item_editor, "update_viewport");
+						undo_redo->add_undo_method(canvas_item_editor, "update_viewport");
+					} else {
+						undo_redo->commit_action();
+					}
```

Comment:
I think here the use of nested actions makes it a bit harder to read especially since we are already checking the exact condition for which we should not be creating this action so I would replace those lines with 
```
					if (action == ACTION_MOVING_POINT) {
						undo_redo->create_action(TTR("Move Point in Curve"));
						undo_redo->add_undo_method(curve.ptr(), "set_point_position", action_point, moving_from);
					}
					undo_redo->add_do_method(curve.ptr(), "set_point_position", action_point, cpoint);
					undo_redo->add_do_method(canvas_item_editor, "update_viewport");
					undo_redo->add_undo_method(canvas_item_editor, "update_viewport");
					undo_redo->commit_action(false);
```

And remove the `update_viewport` on line 163-164 and 201-202 as it is better to have them just in one place instead of three

Reply:
That code doesn't look valid. You are creating a new action inside the condition, but never commit it. A commit has to exist for every started action.